src/main/java/io/fabric8/jenkins/openshiftsync/ConfigMapWatcher.java

```diff
@@ -0,0 +1,202 @@
+
+import com.thoughtworks.xstream.XStreamException;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.triggers.SafeTimerTask;
+import hudson.util.XStream2;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapList;
+import io.fabric8.kubernetes.client.Watcher;
+
+import org.csanchez.jenkins.plugins.kubernetes.PodTemplate;
+import org.csanchez.jenkins.plugins.kubernetes.PodVolumes;
+
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Logger;
+
+import static io.fabric8.jenkins.openshiftsync.OpenShiftUtils.getAuthenticatedOpenShiftClient;
+import static java.util.logging.Level.SEVERE;
+import static java.util.logging.Level.WARNING;
+
+public class ConfigMapWatcher extends BaseWatcher implements Watcher<ConfigMap> {
+    private final Logger logger = Logger.getLogger(getClass().getName());
+    private Map<String, List<PodTemplate>> trackedConfigMaps;
+
+
+    @SuppressFBWarnings("EI_EXPOSE_REP2")
+    public ConfigMapWatcher(String[] namespaces) {
+        super(namespaces);
+        this.trackedConfigMaps = new ConcurrentHashMap<>();
+    }
+    
+    public Runnable getStartTimerTask() {
+        return new SafeTimerTask() {
+            @Override
+            public void doRun() {
+                for(String namespace:namespaces) {
+                    try {
+                        logger.fine("listing ConfigMap resources");
+                        final ConfigMapList configMaps = getAuthenticatedOpenShiftClient().configMaps().inNamespace(namespace).list();
+                        onInitialConfigMaps(configMaps);
+                        logger.fine("handled ConfigMap resources");
+                        if (watches.get(namespace) == null) {
+                            watches.put(namespace,getAuthenticatedOpenShiftClient().configMaps().inNamespace(namespace).withResourceVersion(configMaps.getMetadata().getResourceVersion()).watch(ConfigMapWatcher.this));
+                        }
+                    } catch (Exception e) {
+                        logger.log(SEVERE, "Failed to load ConfigMaps: " + e, e);
+                    }
+                }
+            }
+        };
+    }
+
+    public synchronized void start() {
+        super.start();
+        // lets process the initial state
+        logger.info("Now handling startup config maps!!");
+    }
+
+    @Override
+    public void eventReceived(Action action, ConfigMap configMap) {
+        try {
+            switch (action) {
+                case ADDED:
+                    if(containsSlave(configMap)){
+                        List<PodTemplate> templates = podTemplatesFromConfigMap(configMap);
+                        trackedConfigMaps.put(configMap.getMetadata().getUid(), templates);
+                        for( PodTemplate podTemplate : podTemplatesFromConfigMap(configMap)) {
```

Comment:
why call podTemplatesFromConfigMap again instead of just using the templates var?